spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java

```diff
@@ -57,7 +58,8 @@ public Azure azure() throws IOException {
 
     @Bean
     @ConditionalOnProperty(name = "spring.cloud.azure.telemetryAllowed", havingValue = "true", matchIfMissing = true)
```

Comment:
Could we add another condition for instrument key? Only if it existed do we trigger the telemetry. This is needed when other contributor want to build this project without passing instrument key

Reply:
just for recording here, 2 `ConditionalOnProperty` is not allowed for single bean, and we may need to define new `Conditional` Annotation here. So keep current for now.